hoomd/hpmc/pytest/test_remove_drift.py

```diff
@@ -0,0 +1,189 @@
+
+    # validate the params were set properly
+    for attr, value in constructor_args.items():
+        assert np.all(getattr(remove_drift, attr) == value)
+
+
+@pytest.mark.cpu
+@pytest.mark.parametrize("attr,value", valid_attrs)
+def test_valid_setattr(device, attr, value):
+    """Test that RemoveDrift can get and set attributes."""
+    remove_drift = hoomd.hpmc.update.RemoveDrift(
+        trigger=hoomd.trigger.Periodic(10),
+        reference_positions=[(0, 0, 1), (-1, 0, 1)])
+
+    setattr(remove_drift, attr, value)
+    assert np.all(getattr(remove_drift, attr) == value)
+
+
+@pytest.mark.cpu
+@pytest.mark.parametrize("attr,value", valid_attrs)
+def test_valid_setattr_attached(attr, value, simulation_factory,
+                                two_particle_snapshot_factory, valid_args):
+    """Test that RemoveDrift can get and set attributes while attached."""
+    integrator = valid_args[0]
+    args = valid_args[1]
+    # Need to unpack union integrators
+    if isinstance(integrator, tuple):
+        inner_integrator = integrator[0]
+        integrator = integrator[1]
+        inner_mc = inner_integrator()
+        for i in range(len(args["shapes"])):
+            # This will fill in default values for the inner shape objects
+            inner_mc.shape["A"] = args["shapes"][i]
+            args["shapes"][i] = inner_mc.shape["A"]
+    mc = integrator()
+    mc.shape["A"] = args
+    mc.shape["B"] = args
+
+    remove_drift = hoomd.hpmc.update.RemoveDrift(
+        trigger=hoomd.trigger.Periodic(10),
+        reference_positions=[(0, 0, 1), (-1, 0, 1)])
+    dim = 2 if 'polygon' in integrator.__name__.lower() else 3
+    sim = simulation_factory(
+        two_particle_snapshot_factory(particle_types=['A', 'B'],
+                                      dimensions=dim,
+                                      d=2,
+                                      L=50))
+    sim.operations.updaters.append(remove_drift)
+    sim.operations.integrator = mc
+
+    sim.run(0)
+
+    setattr(remove_drift, attr, value)
+    assert np.all(getattr(remove_drift, attr) == value)
+
+
+@skip_mpi
+@pytest.mark.cpu
+def test_remove_drift(simulation_factory, lattice_snapshot_factory):
+    """Test that RemoveDrift modifies positions correctly."""
+    sim = simulation_factory(
+        lattice_snapshot_factory(particle_types=['A'],
+                                 dimensions=3,
+                                 a=4,
+                                 n=10,
+                                 r=0))
+    sim.seed = 19233
+    mc = hoomd.hpmc.integrate.Sphere(default_d=0.1, default_a=0.1)
+    mc.shape["A"] = dict(diameter=1.0)
+    sim.operations.integrator = mc
+
+    # use initial lattice configuration as reference
+    comm = MPI.COMM_WORLD
+    s = sim.state.snapshot
+    if s.communicator.rank == 0:
+        reference_positions = s.particles.position
+    else:
+        reference_positions = None
+    reference_positions = comm.bcast(reference_positions, root=0)
```

Comment:
Implement the broadcast in C++. Users should not have to use mpi4py to ensure that they specify the same lattice positions on all ranks.

Reply:
This was just a workaround to be used in this test only since I'm not setting the reference positions explicitly on python. Instead, I'm using whatever initial positions are set by the `lattice_snapshot_factory`, which I can only get through the snapshot API. I can avoid it by setting the positions explicitly, I believe.

Reply:
I still think the array should be broadcast in c++ to ensure consistency. Using the snapshot to get the initial lattice is likely a common use-case. Even if users don't use the snapshot, they should not be forced to duplicate the data on all ranks to ensure the simulation is valid.

Reply:
Now that you've added the broadcast in c++, you can remove the broadcast in the python test.